spec/lumberjack_spec.rb

```diff
@@ -76,10 +76,12 @@
     # TODO(sissel): Make sure this doesn't take forever, do a timeout.
     count = 0
     events = []
-    connection = server.accept
-    connection.run do |event|
+    server.run do |event|
       events << event
-      connection.close if events.length == lines.length
+      if events.length == lines.length
+        server.stop
+        raise EOFError.new("Force the connection out of the read loop")
```

Comment:
This seems 'raise to abort a read loop' seems like a bit of a hack. Can we firm this up with an api call or some clearer way to indicate "I let's be done with this" ?